docs/docs/components/textarea.md

```diff
@@ -0,0 +1,169 @@
+textArea.addParagraph("Second paragraph.");
+
+// Fetch and print all paragraphs
+List<String> paragraphs = textArea.getParagraphs();
+System.out.println("Paragraphs: " + paragraphs);
+```
+
+## Line count and paragraph length
+
+The `TextArea` component provides robust controls to ensure input consistency by limiting the number of lines and the length of individual paragraphs. These restrictions are useful in apps where structured text input is essential, such as in form fields, document editors, or any interface requiring precise formatting.
+
+**setLineCountLimit()** - This method specifies the maximum number of lines that can be entered. Once the limit is reached, additional lines are prevented, ensuring the text remains within a controlled vertical space. This helps avoid excessive scrolling and maintains clean UI layouts.
+
+**setParagraphLengthLimit()** - This method enforces a maximum character count per line within a paragraph. This ensures that each line of text remains readable and adheres to a consistent structure. When the character limit for a line is reached, additional input wraps to the next line if line wrapping is enabled. This is especially useful for maintaining clarity and preventing overly long lines of text.
+
+Here’s an example of a feedback form that uses the `TextArea` component to enforce a maximum number of lines and character limits per paragraph, ensuring concise and structured user input:
+
+<ComponentDemo 
+path='/webforj/textarealinelimit?' 
+javaE='https://raw.githubusercontent.com/webforj/webforj-docs-samples/refs/heads/main/src/main/java/com/webforj/samples/views/textarea/TextAreaLineLimitView.java'
+height = '250px'
+/>
+
+## Maximum and minimum length
+
+`TextArea` includes built-in validation features to ensure user input adheres to specific requirements. These features are essential for maintaining data integrity, enhancing user experience, and preventing invalid or incomplete submissions.
+
+The `maxLength` property allows you to define the total number of characters a user can enter. This ensures that input remains concise and within acceptable limits, making it particularly useful for fields like usernames, descriptions, or other constrained text inputs. On the other hand, the `minLength` property enforces a baseline character count, ensuring that users provide enough information before proceeding. This is ideal for fields requiring a minimum level of detail, such as feedback forms or comments.
+
+:::tip Key Difference
+- **Line Count and Paragraph Length**: These focus on structuring text visually. `setLineCountLimit()` restricts the number of lines, while `setParagraphLengthLimit()` limits characters per line, wrapping text if necessary.
+- **Maximum and Minimum Length**: These focus on content validation. `setMaxLength()` and `setMinLength()` control the total number of characters allowed in the TextArea across all lines.
+:::
+
+## Word wrap and line wrapping 
```

Comment:
Shorten the text and replace the demo with something that demonstrates how wrapping works, using a toggle at the bottom of the textarea (not a button). Prefill the textarea with text and add meaningful labels.